### PR TITLE
feat: main NavigationView improvements

### PR DESCRIPTION
**Pull request by huynhsontung**
NavigationView now has a 48px margin from the top, following the WinUI guidelines.

image

Adjusted animation timing for nav bar expand and collapse.

https://github.com/huynhsontung/Screenbox/assets/31434093/3e25f749-778f-41c4-92f1-5200ccf43724

### CODE REVIEW

**Review by United600 — COMMENTED** [review]
We probably shouldn't remove it, otherwise the Grid Headers and NavigationView headers will be different (eg. VideosPage - FolderViewPage) and it'll have cascading effects for alignment.
example

We can add height to the grid row as a fix, I guess.

Edit: Oops, the [commit](https://github.com/huynhsontung/Screenbox/pull/189/commits/244ec4a4376bfe181515cef51797e8aa132b471c#diff-2a65fda2e29f78abdd07a946580e19031e614196813d4484b8ba06beb1c58894).

**Review by United600 — COMMENTED** [review]
Also maybe increase these values (I'm getting -148 for staying the same):
[AlbumArtBG](https://github.com/huynhsontung/Screenbox/pull/189/files#diff-08fca18ca93304b506c122bd2c9e960238d3719c25d281425349070daf58ef24L39)
[AlbumArtBG-Minimal](https://github.com/huynhsontung/Screenbox/pull/189/files#diff-08fca18ca93304b506c122bd2c9e960238d3719c25d281425349070daf58ef24L201)

[ArtistArtBG](https://github.com/huynhsontung/Screenbox/pull/189/files#diff-cb49ec8b9ba7f5eb3dfa816b3499739ea237528ca95d99db89c1895d1abd5969L112)
[ArtistArtBG-Minimal](https://github.com/huynhsontung/Screenbox/pull/189/files#diff-cb49ec8b9ba7f5eb3dfa816b3499739ea237528ca95d99db89c1895d1abd5969L271)

AlbumArt comparison

Well actually more art looks good 👌.

File: Screenbox/Controls/Styles/CustomNavigationView.xaml [hunk 1]
```diff
@@ -0,0 +1,756 @@
...
+                            <Grid
+                                x:Name="ButtonHolderGrid"
+                                Grid.Row="1"
+                                Margin="{ThemeResource NavigationViewButtonHolderGridMargin}">
+                                <Button
+                                    x:Name="NavigationViewBackButton"
+                                    Width="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=TemplateSettings.SmallerPaneToggleButtonWidth}"
+                                    VerticalAlignment="Top"
+                                    IsEnabled="{TemplateBinding IsBackEnabled}"
+                                    Style="{StaticResource NavigationBackButtonNormalStyle}"
```
Comment (United600) [inline Screenbox/Controls/Styles/CustomNavigationView.xaml]: On Windows 10 is using the "wrong style" (in this case the Windows 10 style).
Back button Windows 10

Comment (huynhsontung) [reply Screenbox/Controls/Styles/CustomNavigationView.xaml]: Seems like I used the wrong NavigationView style. Will investigate.

**Comment by huynhsontung** [comment]
Thanks for explaining the LineHeight usage. I wasn't sure why it was there before.

> We can add height to the grid row as a fix, I guess.

Perhaps MinHeight? We allow up to two lines for the page title so we would need the row height to be dynamic.

**Comment by United600** [comment]
> Perhaps MinHeight? We allow up to two lines for the page title so we would need the row height to be dynamic.

I, personally, like LineHeight better. Easier to manage and simplifies our job (don't have to keep track of all headers). Other option is to override the NavigationView height, I did try that but 44 wasn't possible (LineHeight didn't reduce past 45 and something).

File: Screenbox/Controls/Styles/CustomNavigationView.xaml (hunk 1 above)
Comment (United600) [reply Screenbox/Controls/Styles/CustomNavigationView.xaml]: > Seems like I used the wrong NavigationView style. Will investigate.

Also confused, is it this [one](https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/NavigationView/NavigationBackButton_rs1_themeresources.xaml)?

**Comment by United600** [comment]
> Perhaps MinHeight? We allow up to two lines for the page title so we would need the row height to be dynamic.

About that.... I think I removed most of them, only left MaxLines=2 for the search pages.

File: Screenbox/Controls/Styles/CustomNavigationView.xaml (hunk 1 above)
Comment (huynhsontung) [reply Screenbox/Controls/Styles/CustomNavigationView.xaml]: There were more complications than I initially thought with `CustonNavigationViewStyle` so I removed it and went back to modification in code after the template apply.

**Comment by huynhsontung** [comment]
After some discussion, we agreed to use MinHeight to standardize header height everywhere.

**Pull request merged**